Source/WebCore/rendering/RenderLayerBacking.cpp

```diff
@@ -1594,6 +1594,7 @@ void RenderLayerBacking::updateGeometry(const RenderLayer* compositedAncestor)
     LayoutRect ancestorCompositedBounds = compositedAncestor ? compositedAncestor->backing()->compositedBounds() : LayoutRect();
     setRequiresOwnBackingStore(compositor().requiresOwnBackingStore(m_owningLayer, compositedAncestor,
         LayoutRect(toLayoutPoint(compositedBoundsOffset.fromParentGraphicsLayer()), compositedBounds().size()), ancestorCompositedBounds));
+    updateDrawsContent();
```

Comment:
This is called from `updateAfterDescendants()` which is called on every RenderLayerBacking from `RenderLayerCompositor::updateBackingAndHierarchy()` so why does calling it here change behavior?

I'm also curious about why this change is needed on GTK but not macOS/iOS.

Reply:
GTK is the only port that doesn't use force compositing mode by default.
When `RenderLayerCompositor::updateCompositingLayers()` is called the first time with force compositing enabled, we create the root render layer. It will have no children yet. But its `drawsContent` flag will be set to true by the end of the method. 
But with force compositing disabled, we create the root layer (with `drawsContent` set to false by default) when we already have some children layers. The first child will then decide that it needs its own backing store. And only after that, the root layer will set `drawsContent` to true. But it may be too late.
I hope I explained it well enough.

Reply:
Another argument for this change is that the returning value of `RenderLayerBacking::containsPaintedContent()`, which is used by `RenderLayerBacking::updateDrawsContent()`, depends on its `m_requiresOwnBackingStore` value. So, whenever we change it, we should call `updateDrawsContent()`